core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -130,6 +130,12 @@ private[spark] class Executor(
   private val urlClassLoader = createClassLoader()
   private val replClassLoader = addReplClassLoaderIfNeeded(urlClassLoader)
 
+  Thread.currentThread().setContextClassLoader(replClassLoader)
```

Comment:
do we want to do it in the same thread?  It might be safer in a separate thread. Does that affect your memory monitor?

Reply:
I was thinking another thread would at least prevent them from not allowing the executor to run/start.  If someone added a plugin that just blocked or did something that took time and then you started to see timeouts during start, those might not be as obvious what is going on.  If we start it in a separate thread, yes it still uses resources but it doesn't completely block the executor from starting and trying to take tasks.  It also just seems safer to me as you could try to catch exceptions from there and possibly ignore them so it doesn't affect the main running.

Reply:
makes sense.  probably something I should have at least a basic test for as well ... will need to think about how to do that.